extensions/interactions/ItemSelectionInput/directives/ItemSelectionInputRulesServiceSpec.js

```diff
@@ -72,4 +72,18 @@ describe('Item Selection rules service', function() {
         ['a', 'b', 'c', 'd'], RULE_INPUT)).toBe(false);
     }
   );
+  it('should have a correct \'is a proper subset of\' rule',
+    function() {
+      var RULE_INPUT = {
+        x: ['a', 'b', 'c']
+      };
+      expect(isirs.IsProperSubsetOf(['a'], RULE_INPUT)).toBe(true);
+      expect(isirs.IsProperSubsetOf(['a', 'b'], RULE_INPUT)).toBe(true);
+      expect(isirs.IsProperSubsetOf(['a', 'b', 'c'], RULE_INPUT)).toBe(false);
+      expect(isirs.IsProperSubsetOf(
+        ['a', 'b', 'c', 'd'], RULE_INPUT)).toBe(false);
+      expect(isirs.IsProperSubsetOf(
+        ['b', 'c', 'd', 'e'], RULE_INPUT)).toBe(false);
```

Comment:
Add one more test to ensure empty set is also considered as a proper subset.

Reply:
@bansalnitish , Done, Thanks for reviewing!

Reply:
and also one more case where the answer is not in the subset at all.

Reply:
@amey-kudari did you miss this comment?

Reply:
@seanlip I added this test
```javascript
expect(isirs.IsProperSubsetOf(['d', 'e'], RULE_INPUT)).toBe(false);
```
in line 84

Reply:
Yes, but you need to reply to the comment too.

Reply:
Yeah, sorry I will make sure I reply to all reviews individually (I replied in the end saying l did all of them, https://github.com/oppia/oppia/pull/6350#issuecomment-471248212)